jenkins/check-for-build.jenkinsfile

```diff
@@ -0,0 +1,89 @@
+lib = library(identifier: "jenkins@20211123", retriever: legacySCM(scm))
+
+pipeline {
+    agent none
+    triggers {
+        parameterizedCron '''
+            H/10 * * * * %INPUT_MANIFEST=1.3.0/opensearch-1.3.0.yml;TARGET_JOB_NAME=distribution-build-opensearch
+            H/10 * * * * %INPUT_MANIFEST=1.3.0/opensearch-dashboards-1.3.0.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards
+            H 1 * * * %INPUT_MANIFEST=2.0.0/opensearch-2.0.0.yml;TARGET_JOB_NAME=distribution-build-opensearch
+            H 1 * * * %INPUT_MANIFEST=2.0.0/opensearch-dashboards-2.0.0.yml;TARGET_JOB_NAME=distribution-build-opensearch-dashboards
+        '''
+    }
+    parameters {
+        string(
+            name: 'INPUT_MANIFEST',
+            description: 'Input manifest under the manifests folder, e.g. 2.0.0/opensearch-2.0.0.yml.',
+            trim: true
+        )
+        string(
+            name: 'TARGET_JOB_NAME',
+            description: 'Job to trigger if build has changed',
+            trim: true
+        )
+    }
+    stages {
+        stage('detect docker image + args') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211028'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script { 
+                    dockerAgent = detectDockerAgent()
+                }
+            }
+        }
+        stage('trigger-build-if-needed') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-arm64-c6g4xlarge-Docker-Host'
+                    image dockerAgent.image
+                    args dockerAgent.args
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    lock(resource: "CheckForBuild-${INPUT_MANIFEST}-${TARGET_JOB_NAME}", skipIfLocked: true) {
+                        def sha = getManifestSHA(jobName: "${TARGET_JOB_NAME}")
+                        if (sha.exists) {
+                            echo "Skipping, ${sha.path} already exists."
+                        } else {
+                            build job: "${TARGET_JOB_NAME}", parameters: [
+                                string(name: 'INPUT_MANIFEST', value: "${INPUT_MANIFEST}")
+                            ], wait: true
+                            echo "Build succeeded, uploading build SHA for that job"
```

Comment:
Don't we want a notification message when the actual build is successful?

Reply:
I've delegated detailed notifications for the build jobs themselves as they will have all the relevant context,  The failure-only notification on this job is to make sure that if we have a failure checking/trigger for a build job it doesn't go unnoticed which could prevent the downstream error message from appearing

Reply:
Ahh! We have notification for build job. Forgot that, Thanks for reminding!